sdk/lib/event_group/event_group.cc

```diff
@@ -42,16 +43,16 @@ int eventgroup_create(Timeout            *timeout,
 	{
 		return -ERANGE;
 	}
-	size_t size = sizeof(EventGroup) + (threads * sizeof(EventWaiter));
-	auto   group =
-	  static_cast<EventGroup *>(heap_allocate(timeout, heapCapability, size));
-	*outGroup = group;
-	if (!__builtin_cheri_tag_get(group))
-	{
-		return -ENOMEM;
-	}
-	group->waiterCount = threads;
-	return 0;
+	size_t size  = sizeof(EventGroup) + (threads * sizeof(EventWaiter));
+	auto   alloc = heap_allocate_cpp(timeout, heapCapability, size);
+	return alloc.either(
```

Comment:
I wonder if this (`.either([...](auto c){ ...; return 0; }, [](int e){ return e; })`) is enough of a pattern that we should give it a name?  (Basically a twist on `.and_then(f)` with `f` returning `void`?)

Reply:
Yes, I was thinking the same. It only applies if you are converting from `ErrorOr` to `int` success/fail but potentially useful.